### PR TITLE
Set up pipeline on cloud, added ecr URI, updated etl for lambda_handler

### LINKED ISSUE #72: Setup ecr and deploy pipeline to cloud
**Description:**
Set up an Amazon Elastic Container Registry (ECR) to store container images and create a deployment pipeline to automate the process of building, testing, and deploying these containers to the cloud.

**Outcome:**
A fully configured ECR repository and a deployment pipeline that automates the steps of building Docker images, pushing them to ECR, and deploying them to the cloud environment.

**User Story:**
As a DevOps engineer, I need to set up an ECR and deploy pipeline, so we can efficiently manage containerized applications and automate their deployment to the cloud.

**Design:**

    ECR Setup:
-         Create an ECR repository for storing Docker images.
-         Configure access permissions for pushing and pulling images.
   Deployment Pipeline:
-         Automate image building using CI/CD tools (e.g., GitHub Actions, Jenkins).
-         Push the built images to the ECR repository.
-         Deploy the images to a cloud environment (e.g., ECS, EKS) with necessary infrastructure defined in 
-         Test the pipeline end-to-end to verify seamless deployments.

### PR DESCRIPTION
**Pull request by qrafiq1**
Had to remove asyncio from requirements.txt for lambda AWS to work, this needs to be noted down in the README.

main() function in etl.py was removed and directly called in lambda_handler for async.

Closes #72 .

### CODE REVIEW

File: pipeline/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 COPY requirements.txt .
 RUN pip3 install -r requirements.txt
 
+COPY extract.py .
+COPY transform.py .
+COPY load.py .
 COPY etl.py .
+COPY extract.py .
+COPY transform.py .
+COPY load.py .
```
Comment (FarihaChoudhury) [inline pipeline/Dockerfile:14]: why are the same files being copied twice?

**Review by keoghrmj — APPROVED** [review]
Fix what Fariha said but apart from that, looking good

File: terraform/main.tf [hunk 2]
```diff
@@ -68,6 +68,12 @@ resource "aws_iam_role_policy" "c14-runtime-terrors-plants-etl-lambda_execution_
   })
 }
 
+resource "aws_cloudwatch_log_group" "lambda_log_group" {
+  name              = "/aws/lambda/c14-runtime-terrors-plants-etl-lambda-function-tf"
+  retention_in_days = 7
+}
```
Comment (FarihaChoudhury) [inline terraform/main.tf:74]: im sorry for missing this :(

File: terraform/main.tf [hunk 3]
```diff
@@ -153,249 +159,249 @@ resource "aws_scheduler_schedule" "c14-runtime-terrors-plants-etl-schedule-tf" {
...
+
+
+# }
+
+# resource "aws_ecs_service" "c14-runtime-terrors-plants-dashboard-service-tf" {
+#     name            = "c14-runtime-terrors-plants-dashboard-service-tf"
+#     cluster         = data.aws_ecs_cluster.c14-cluster.id
+#     task_definition = aws_ecs_task_definition.c14-runtime-terrors-plants-dashboard-ECS-task-def-tf.arn
+#     desired_count   = 1
+#     launch_type     = "FARGATE"
```
Comment (SurinaCS) [inline terraform/main.tf:400]: is this all meant to be commented out

**Review by SurinaCS — COMMENTED** [review]
great just uncomment

File: terraform/main.tf (hunk 2 above)
Comment (qrafiq1) [reply terraform/main.tf:74]: is okay

File: terraform/main.tf (hunk 3 above)
Comment (qrafiq1) [reply terraform/main.tf]: fixed it, thank you.

**Pull request merged**